src/dvclive/live.py

```diff
@@ -191,6 +198,78 @@ def log_plot(self, name, labels, predictions, **kwargs):
         data.dump(val, self._step, **kwargs)
         logger.debug(f"Logged {name}")
 
+    def log_params(
+        self, params: Dict[str, ParamsLike], name: str = PARAMS_DEFAULT_NAME
```

Comment:
What about handling nested dicts? Similarly, can `log_param("train.lr", 0.1)` be saved as a nested dict?

Reply:
We should be handling those, although I think this can become tricky: think of more complex objects, such as an optimizer (like in here https://keras.io/api/optimizers/). How are we supposed to capture these kind of parameters?

Reply:
I would expect people to log that kind of thing as follows:

`Live().log_params("train", {"lr": 0.01})`

Which works, although I don't know if unexpectedly 😅 as it doesn't match the `ParamsLike` type @dtrifiro

Reply:
Yep, `ParamsLike` should really look like this: `ParamsLike = Union[int, float, str, bool, "ParamsLike"]`, although mypy complains because of a cyclic definition. Will check this out later.

Reply:
Sorry, for some reason I thought the dot-notation was supported for logging metrics. That shouldn't be necessary, but handling nested dicts should be.

Some interesting reading:
* https://stackoverflow.com/questions/68167770/type-hints-for-variably-nested-dict-in-python
* https://github.com/python/mypy/issues/731

Reply:
Fixed `ParamsLike` to include the recursive definition. For now we support dumping any parameters that are supported by `json.dump`. Complex (custom) parameters are still not supported. Do we want to support those right away, or can we leave those to another iteration?